lib/policies/exponentialRetryPolicy.ts

```diff
@@ -150,7 +150,9 @@ function retry(policy: ExponentialRetryPolicy, request: WebResource, response: H
     // If the operation failed in the end, return all errors instead of just the last one
     requestError = retryData.error;
     return Promise.reject(requestError);
-  } else {
+  } else if (response) {
     return Promise.resolve(response);
+  } else {
+    return Promise.reject();
```

Comment:
needs to reject with an actual error value in the case that there was no response and also no error (either that or use `response!` on 154)